astropy/coordinates/tests/test_sky_coord.py

```diff
@@ -610,6 +620,22 @@ def test_position_angle_directly():
     assert result.value == 0.
 
 
+def test_directional_offset_by():
```

Comment:
Can you add at least a couple of tests which are "absolute", i.e. you hardcode an offset_by or offset_to and the expected answer.  The current tests are a good way of running a bunch of cases but there could, in theory, be offsetting errors so that the round-trip always works.  I would suggest one test that starts at lon,lat = 0, 89 and does a 2 deg offset straight over the pole (posangle=90?) and winds up at 180, 89.